### PR TITLE
Streamer shelves

### PR DESCRIPTION
**Pull request by PSYmoom**

### CODE REVIEW

File: components/twitchAnnouncement.py [hunk 1]
```diff
@@ -1,73 +1,155 @@
...
-                'Authorization': 'Bearer ' + twitch_OAuth_token
-            }
+    try:
+        # Need to generate OAuth token on startup and failed generation
+        if not twitch_OAuth_token:
+            await generate_twitch_OAuth_token(channel)
 
-            stream = requests.get('https://api.twitch.tv/helix/streams' + twitch_streamers, headers=headers)
-            stream = stream.json();
+        streamer_list_shelf = shelve.open(CONST_STREAMER_DB_PATH)
```
Comment (mmaaz1) [inline components/twitchAnnouncement.py]: nit: no need to call it const

File: components/twitchAnnouncement.py [hunk 2]
```diff
@@ -1,73 +1,155 @@
...
+        # If livestream is new, create message
+        # else update the viewer count
+        fetched_livestreams = {}
+        for livestream in stream.get('data'):
+            user_name = livestream['user_name']
+            if user_name not in twitch_curr_livestreams:
+                message = await channel.send(generate_message(livestream))
+            else:
+                message = await twitch_curr_livestreams[user_name].edit(content=generate_message(livestream))
+            print(livestream)
```
Comment (mmaaz1) [inline components/twitchAnnouncement.py]: nit: print

File: components/twitchAnnouncement.py [hunk 3]
```diff
@@ -1,73 +1,155 @@
...
+    try:
+        result = streamerDatabase.remove_item(message, admin_role)
+        await ut.send_message(channel, result)
+    except Exception as e:
+        await ut.send_message(channel, "Error Removing Streamer: " + str(e))
+
+
+# Lists the currently tracked streamers. Does not need admin access.
+async def list_streamers(channel):
+    result = streamerDatabase.list_items()
```
Comment (mmaaz1) [inline components/twitchAnnouncement.py]: nit: we should wrap user command handlers with try excepts.

**Review by mmaaz1 — DISMISSED** [review]

File: components/twitchAnnouncement.py (hunk 3 above)
Comment (PSYmoom) [reply components/twitchAnnouncement.py]: Added try except.

File: components/twitchAnnouncement.py (hunk 2 above)
Comment (PSYmoom) [reply components/twitchAnnouncement.py]: Removed print statement,

File: components/twitchAnnouncement.py (hunk 1 above)
Comment (PSYmoom) [reply components/twitchAnnouncement.py]: CONST_ removed.

**Pull request merged**